src/concurrency/data_race.rs

```diff
@@ -361,6 +363,8 @@ impl MemoryCellClocks {
             atomic.read_vector.set_at_index(&clocks.clock, index);
             Ok(())
         } else {
+            let atomic = self.atomic_mut();
+            atomic.read_vector.set_at_index(&clocks.clock, index);
```

Comment:
This is duplicated with the previous `if` arm. Can that be avoided? (Happens a few more times below.)

Reply:
I think I got all of them?